src/scripts/main.js

```diff
@@ -1,3 +1,344 @@
+}
+
+function slideLeft() {
+  let isAnyCellMoved = false;
+
+  for (let r = 0; r < rows; r++) {
+    let row = board[r];
+    const prevRow = [...row];
+
+    row = slide(row);
+    board[r] = row;
+
+    if (!arraysAreEqual(prevRow, row)) {
+      isAnyCellMoved = true;
+    }
+
+    for (let c = 0; c < columns; c++) {
+      const fieldCell = document.getElementById(r.toString() + '-' + c.toString());
+      const cellValue = board[r][c];
+
+      updateFieldCell(fieldCell, cellValue);
+    }
+  }
+
+  if (isAnyCellMoved) {
+    addRandomCell();
+  }
+}
+
+function slideRight() {
+  let isAnyCellMoved = false;
+
+  for (let r = 0; r < rows; r++) {
+    let row = board[r];
+    const prevRow = [...row];
+
+    row.reverse();
+
+    row = slide(row);
+    row.reverse();
+    board[r] = row;
+
+    if (!arraysAreEqual(prevRow, row)) {
+      isAnyCellMoved = true;
+    }
+
+    for (let c = 0; c < columns; c++) {
+      const fieldCell = document.getElementById(r.toString() + '-' + c.toString());
+      const cellValue = board[r][c];
+
+      updateFieldCell(fieldCell, cellValue);
+    }
+  }
+
+  if (isAnyCellMoved) {
+    addRandomCell();
+  }
+}
+
+function slideUp() {
+  let isAnyCellMoved = false;
+
+  for (let c = 0; c < columns; c++) {
+    let row = [board[0][c], board[1][c], board[2][c], board[3][c]];
+    const prevRow = [...row];
+
+    row = slide(row);
+
+    if (!arraysAreEqual(prevRow, row)) {
+      isAnyCellMoved = true;
+    }
+
+    for (let r = 0; r < rows; r++) {
+      board[r][c] = row[r];
+
+      const fieldCell = document.getElementById(r.toString() + '-' + c.toString());
+      const cellValue = board[r][c];
+
+      updateFieldCell(fieldCell, cellValue);
+    }
+  }
+
+  if (isAnyCellMoved) {
+    addRandomCell();
+  }
+}
+
+function slideDown() {
+  let isAnyCellMoved = false;
+
+  for (let c = 0; c < columns; c++) {
+    let row = [board[0][c], board[1][c], board[2][c], board[3][c]];
+    const prevRow = [...row];
+
+    row.reverse();
+    row = slide(row);
+    row.reverse();
+
+    if (!arraysAreEqual(prevRow, row)) {
+      isAnyCellMoved = true;
+    }
+
+    for (let r = 0; r < columns; r++) {
+      board[r][c] = row[r];
+
+      const fieldCell = document.getElementById(r.toString() + '-' + c.toString());
+      const cellValue = board[r][c];
+
+      updateFieldCell(fieldCell, cellValue);
+    }
+  }
+
+  if (isAnyCellMoved) {
+    addRandomCell();
+  }
+}
+
+function gameIsOver() {
+  const hasNoEmptyCells = !hasEmptyFieldCell();
+  let move = true;
+
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      if (c > 0 && board[r][c] === board[r][c - 1]) {
+        move = false;
+      } else if (c < columns - 1 && board[r][c] === board[r][c + 1]) {
+        move = false;
+      } else if (r < rows - 1 && board[r][c] === board[r + 1][c]) {
+        move = false;
+      } else if (r > 0 && board[r][c] === board[r - 1][c]) {
+        move = false;
+      }
+    }
+  }
```

Comment:
instead of writing else if use keyword break to exit loop in each if brunch. It will keep you code clean.

```javascript
if (condition) {
move = false;
break;
}

if (condition) {
move = false;
break;
}

// and so on
``` 

or you can even combine those conditions since you're doing the same action.

```javascript
//moving conditions into variables for redability
const condition1 = // your code;
const condition2 = // your code;

if (condition1 || condition2 || //...) {
move = false;
}
```